Bump main version to 9.0-alpha1

carlossanlop opened the pull request:
🎉

carlossanlop committed:
Bump main version to 9.0-alpha1

carlossanlop commented:
@dotnet/ncl or @ViktorHofer do you know how to address the Quic failure message about opting into preview features?:

```
src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3Connection.cs(29,37): error CA2252: (NETCORE_ENGINEERING_TELEMETRY=Build) '_activeRequests''s type contains the preview type 'QuicStream' and requires opting into preview features. See https://aka.ms/dotnet-warnings/preview-features for more information. (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2252)
```

wfurt commented:
Strange, HTTP already has the preview feature set:

https://github.com/dotnet/runtime/blob/333c2c7eb28b1080aea379dab801778d3e992609/src/libraries/System.Net.Http/src/System.Net.Http.csproj#L6

Perhaps something in the SDK? 

BTW This may also break the Enterprise pipeline  - at least it always did in the past.

jkoritzinsky commented:
EnablePreviewFeatures has an interesting behavior that it only works on the most current TFM. Looks like we might need some other solution here.

wfurt commented:
removing `RequiresPreviewFeaturesAttribute` from `Quic` for now? I  would hope nobody would use 9.0 in production yet.

jkoritzinsky commented:
Maybe manually including the attribute in the .NET 8 build (instead of using the SDK property) would work.

lewing committed:
Try bumping SdkBandVersion

antonfirsov commented:
/azp-run runtime-libraries stress-http

antonfirsov commented:
/azp-run runtime-libraries enterprise-linux

lewing committed:
Bump the emscripten transport package

lewing committed:
Fix typo

jkoritzinsky commented:
@carlossanlop @wfurt I looked more into the failure. Here's whats happening:

- We have infrastructure to automatically say that for `<MajorVersion>N</MajorVersion>`, the latest TFM is `netN.0`.
- This PR updates `MajorVersion` to `9`.
- This PR does not update `NetCoreAppCurrent` to `net9.0`.
- As a result, System.Net.Http is building for `net8.0`.
- The SDK isn't applying the `EnablePreviewFeatures` attribute as `net8.0` is no longer the "latest TFM".

There are two possible solutions:

- Manually include the `EnablePreviewFeatures` attribute until we change `NetCoreAppCurrent`.
- Change `NetCoreAppCurrent` (and `NetCoreAppMinimum` and `NetCoreAppPrevious` as well) now and face the problems that come with that now.

@carlossanlop which would you rather do? I'd prefer the second option personally.

jkoritzinsky commented:
I've started down the path of the second option locally. I'll try to finish it up tomorrow so I can push it out (just fixing up APICompat baselining).

jkoritzinsky committed:
Update TFM to net9.0, minimum to net8.0, use RC1 nightly for SDK to avoid regressions in .NET 8.0 targets and clean up .NET 8->.NET 9 API compat baselining.

jkoritzinsky committed:
Add compat suppressions for removed EOL TFM assets that will not be in the 9.0 variants of the packages.

ViktorHofer commented on src/coreclr/System.Private.CoreLib/CompatibilitySuppressions.xml:
```diff
@@ -1,45 +1,46 @@
     <Target>M:System.Diagnostics.Tracing.EventSource.Write``1(System.String,``0):[T:System.Diagnostics.CodeAnalysis.RequiresUnreferencedCodeAttribute]</Target>
-    <Left>ref/net8.0/System.Private.CoreLib.dll</Left>
-    <Right>lib/net8.0/System.Private.CoreLib.dll</Right>
+    <Left>ref/net9.0/System.Private.CoreLib.dll</Left>
+    <Right>lib/net9.0/System.Private.CoreLib.dll</Right>
```
Feel free to remove all left and right entries from this file to avoid hardcoding the TFM. We usually only keep left and right when the suppression file applies to multiple assembly identities.

ViktorHofer commented on src/libraries/Microsoft.Extensions.DependencyInjection.Specification.Tests/src/CompatibilitySuppressions.xml:
```diff
@@ -1,7 +1,6 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <!-- https://learn.microsoft.com/en-us/dotnet/fundamentals/package-validation/diagnostic-ids -->
 <Suppressions xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
-  <!-- Test renamed to TypeActivatorCreateInstanceUsesLongestAvailableConstructor. -->
```
undo

ViktorHofer commented on src/libraries/Microsoft.Extensions.DependencyInjection/src/CompatibilitySuppressions.xml:
```diff
@@ -1,7 +1,6 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <!-- https://learn.microsoft.com/en-us/dotnet/fundamentals/package-validation/diagnostic-ids -->
 <Suppressions xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
-  <!-- RequiresDynamicCodeAttribute was removed during .NET 8 with https://github.com/dotnet/runtime/commit/f424c0f7ce8b1453d7124a3def8cf44a4ff42ead. -->
```
undo

ViktorHofer commented on src/libraries/Microsoft.Extensions.Hosting/src/CompatibilitySuppressions.xml:
```diff
@@ -1,7 +1,6 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <!-- https://learn.microsoft.com/en-us/dotnet/fundamentals/package-validation/diagnostic-ids -->
 <Suppressions xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
-  <!-- RequiresDynamicCode attributes removed during .NET 8: https://github.com/dotnet/runtime/commit/f424c0f7ce8b1453d7124a3def8cf44a4ff42ead. -->
```
undo

ViktorHofer commented on src/libraries/Microsoft.Extensions.Http/src/Microsoft.Extensions.Http.csproj:
```diff
@@ -33,7 +33,7 @@ System.Net.Http.IHttpClientFactory</PackageDescription>
   </ItemGroup>
 
-  <ItemGroup Condition="'$(TargetFramework)' == 'net8.0'">
+  <ItemGroup Condition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net8.0'))">
```
Please prefer `TFI == .NETCoreApp` when `NetCoreAppMinimum` is equal or greater than the min version in the condition. The MSBuild functions aren't for free.

ViktorHofer commented on src/libraries/Microsoft.Extensions.Logging/src/CompatibilitySuppressions.xml:
```diff
@@ -1,7 +1,6 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <!-- https://learn.microsoft.com/en-us/dotnet/fundamentals/package-validation/diagnostic-ids -->
 <Suppressions xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
-  <!-- RequiresDynamicCodeAttribute was removed during .NET 8 with https://github.com/dotnet/runtime/commit/f424c0f7ce8b1453d7124a3def8cf44a4ff42ead. -->
```
undo

ViktorHofer commented on src/libraries/System.Security.Cryptography.Xml/src/CompatibilitySuppressions.xml:
```diff
@@ -3,7 +3,6 @@
 <Suppressions xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
   <Suppression>
     <DiagnosticId>CP0001</DiagnosticId>
-    <!-- Type must be public in implementation for serialization to work but we don't want to expose it publicly in the contract as it isn't public on .NET Framework. -->
```
undo

ViktorHofer commented on src/libraries/System.Reflection.Context/src/CompatibilitySuppressions.xml:
```diff
@@ -1,8 +1,22 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
+<!-- https://learn.microsoft.com/en-us/dotnet/fundamentals/package-validation/diagnostic-ids -->
 <Suppressions xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
-  <!-- The assembly version intentionally doesn't match between contract and implementation. -->
```
undo

ViktorHofer commented on src/libraries/System.IO.Pipes/src/CompatibilitySuppressions.xml:
```diff
@@ -1,46 +1,88 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
+<!-- https://learn.microsoft.com/en-us/dotnet/fundamentals/package-validation/diagnostic-ids -->
 <Suppressions xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
-  <!-- Exposed public in System.IO.Pipes.AccessControl but implemented in System.IO.Pipes. -->
```
undo

ViktorHofer commented on Directory.Build.targets:
```diff
@@ -49,6 +49,13 @@
     <SupportedNETCoreAppTargetFramework Include=".NETCoreApp,Version=v$(NETCoreAppMaximumVersion)" DisplayName=".NET $(NETCoreAppMaximumVersion)" Alias="net$(NETCoreAppMaximumVersion)" />
   </ItemGroup>
 
+  <!-- Add known packs from the downlevel TFM until we are building with an SDK that supports the new TFM all-up. -->
+  <ItemGroup Condition="'$(NetCoreAppToolCurrent)' != '$(NetCoreAppCurrent)'">
+    <KnownILLinkPack Include="@(KnownILLinkPack->WithMetadataValue('Identity', 'Microsoft.NET.ILLink.Tasks')->WithMetadataValue('TargetFramework', '$(NetCoreAppToolCurrent)'))">
+      <TargetFramework>$(NetCoreAppCurrent)</TargetFramework>
+    </KnownILLinkPack>
+  </ItemGroup>
```
Please put this into targetingpacks.targets

jkoritzinsky committed:
PR feedback

jkoritzinsky committed:
Fix ILLink errors in the build.

lewing committed:
Update Versions.props

lewing commented:
```
.dotnet/sdk/8.0.100-rc.1.23415.11/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(90,5): error NETSDK1211: (NETCORE_ENGINEERING_TELEMETRY=Restore) EnableSingleFileAnalyzer is not supported for the target framework. Consider multi-targeting to a supported framework to enable single-file analysis, and set EnableSingleFileAnalyzer only for the supported frameworks. For example:
<EnableSingleFileAnalyzer Condition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net6.0'))">true</EnableSingleFileAnalyzer>
```

ViktorHofer commented on src/coreclr/nativeaot/System.Private.CoreLib/src/CompatibilitySuppressions.xml:
```diff
@@ -959,43 +959,43 @@
     <Target>M:System.Diagnostics.Tracing.EventSource.Write``1(System.String,``0):[T:System.Diagnostics.CodeAnalysis.RequiresUnreferencedCodeAttribute]</Target>
-    <Left>ref/net8.0/System.Private.CoreLib.dll</Left>
-    <Right>lib/net8.0/System.Private.CoreLib.dll</Right>
+    <Left>ref/net9.0/System.Private.CoreLib.dll</Left>
+    <Right>lib/net9.0/System.Private.CoreLib.dll</Right>
```
Remove these lefts and rights from this file as well.

ViktorHofer commented on src/coreclr/crossgen-corelib.proj:
```diff
@@ -158,4 +158,6 @@
     <Message Importance="High" Text="Product binaries are available at $(BinDir)" />
   </Target>
+
+  <Import Project="$(RepositoryEngineeringDir)targetingpacks.targets" />
```
Why is that import needed in a NoTargets SDK project?

ViktorHofer commented on src/coreclr/runtime.proj:
```diff
@@ -6,6 +6,7 @@
 
     <_BuildNativeTargetOS>$(TargetOS)</_BuildNativeTargetOS>
     <_BuildNativeTargetOS Condition="'$(TargetsLinuxBionic)' == 'true'">linux-bionic</_BuildNativeTargetOS>
+    <RunAnalyzers>false</RunAnalyzers>
```
Why is this necessary? It would be good to set RunAnalyzers to false globally for NoTargets SDK projects. See https://github.com/dotnet/runtime/blob/296a1d5b9fcea725d3ba7c7b5991885a8f99dbd2/Directory.Build.props#L410

jkoritzinsky replied:
We enable some of the "Form Factor" analyzers like the Trim analyzer and the SingleFile analyzer globally in the repo. Now the SDK warns if you try to enable any of these analyzers and you don't have an ILLink pack available for the target TFM. As a result, it's tripping on all of our NoTargets projects. I'll move this check up to there.

ViktorHofer commented on src/coreclr/tools/aot/ILCompiler.Build.Tasks/ILCompiler.Build.Tasks.csproj:
```diff
@@ -9,12 +9,16 @@
     <PackageReference Include="Microsoft.Build.Utilities.Core" Version="$(MicrosoftBuildUtilitiesCoreVersion)" />
     <PackageReference Include="System.Reflection.Metadata" Version="$(SystemReflectionMetadataVersion)" />
   </ItemGroup>
-
```
Please undo this deletion. We aim towards having our project files represent the `dotnet new` ones which include a new line after the project start tag and before the project close tag.

ViktorHofer commented on src/coreclr/tools/aot/ILCompiler.Build.Tasks/ILCompiler.Build.Tasks.csproj:
```diff
@@ -9,12 +9,16 @@
     <OutputPath>$(RuntimeBinDir)/ilc-published/netstandard</OutputPath>
     <Configurations>Debug;Release;Checked</Configurations>
     <PlatformTarget>AnyCPU</PlatformTarget>
+    <!--
+      The SDK warns if a project targets a framework that doesn't support trimming and IsTrimmable is true.
+      We set IsTrimmable higher up in the Directory.Build.props files, so we reset it to false here.
+    -->
+    <IsTrimmable>false</IsTrimmable>
```
So netstandard2.0 doesn't support trimming, at all? If so, should we set this globally in a targets file based on the TFM?

jkoritzinsky replied:
Yes, we don't have trimming support for .NET Standard 2.0 (it's completely unannotated, so we can't have accurate analysis).

ViktorHofer replied:
Then shouldn't we disable this globally in a targets file for all projects?

ManickaP commented:
What about just `<NoWarn>CA2252</NoWarn>`?

jkoritzinsky committed:
PR feedback

jkoritzinsky committed:
Don't use non-portable rids in src/tests infra

jkoritzinsky committed:
Merge branch 'main' of github.com:dotnet/runtime into BumpMainTo90Alpha1

jkoritzinsky committed:
Fix restore for Microsoft.NETCore.App.Runtime-based sfxprojs

jkoritzinsky committed:
Fix libs build and expand where we set RunAnalyzers

ViktorHofer commented on src/coreclr/tools/aot/Directory.Build.targets:
```diff
@@ -1,6 +1,6 @@
   <PropertyGroup>
     <IsTrimmable Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp'">true</IsTrimmable>
   </PropertyGroup>
+  <Import Project="../Directory.Build.targets" />
```
Please move the import above the custom property as we do in other Directory.Build.targets files: https://github.com/dotnet/runtime/blob/64243bbf5e9ee53c0c4c5678f2cd8c7f1c9b4f6f/src/libraries/System.Runtime.Caching/Directory.Build.targets#L2

ViktorHofer commented on src/installer/pkg/sfx/Directory.Build.props:
```diff
@@ -10,6 +10,8 @@
     <ProductBrandPrefix>Microsoft .NET</ProductBrandPrefix>
     <DisableImplicitFrameworkReferences>true</DisableImplicitFrameworkReferences>
     <UseRuntimePackageDisclaimer>true</UseRuntimePackageDisclaimer>
+    <!-- We don't have any sources here, so don't run analyzers. -->
+    <RunAnalyzers>false</RunAnalyzers>
```
You could also condition on shared framework packages with the following property: https://github.com/dotnet/runtime/blob/64243bbf5e9ee53c0c4c5678f2cd8c7f1c9b4f6f/Directory.Build.props#L408

jkoritzinsky replied:
I can't as that property isn't set early enough to be available in Directory.Build.props files. I tried it and it didn't work.

ViktorHofer replied:
Consider setting it in analyzers.targets. At that point, the property should already be set.

ViktorHofer commented on Directory.Build.props:
```diff
@@ -429,6 +429,11 @@
     <PublishWindowsPdb>false</PublishWindowsPdb>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(UsingMicrosoftNoTargetsSdk)' == 'true'">
+    <!-- Explicitly disable running analyzers to avoid trying to discover the correct ILLink tool pack for a NoTargets project. -->
+    <RunAnalyzers>false</RunAnalyzers>
+  </PropertyGroup>
```
nit: can you please move this into Analyzers.targets? We already have some conditions there to determine when to not run analyzers.

jkoritzinsky committed:
PR feedback

jkoritzinsky committed:
Try fixing the package testing.

jkoritzinsky committed:
Fix remaining restore issues.

jkoritzinsky committed:
Patch preserialized resources to account for the major version bump (convert Version=8.0.0.0 to Version=9.0.0.0)

jkoritzinsky committed:
Patch preserialized resources to account for the major version bump (convert Version=8.0.0.0 to Version=9.0.0.0)

carlossanlop commented on Directory.Build.props:
```diff
@@ -80,20 +80,20 @@
   <!-- The TFMs to build and test against. -->
   <PropertyGroup>
-    <NetCoreAppCurrentVersion>8.0</NetCoreAppCurrentVersion>
+    <NetCoreAppCurrentVersion>9.0</NetCoreAppCurrentVersion>
```
This wasn't supposed to happen in this PR, right? @ViktorHofer

jkoritzinsky replied:
Basically we had the option to either work around the fact that we weren't doing this with the EnablePreviewFeatures work or just doing it and getting it over with.

I decided to do it as I didn't see a reason to work around the EnablePreviewFeatures issue instead of just fixing it and I thought it would be a much easier change than it ended up being.

If we want this to be in a separate branch, I can branch off from here with those changes in a draft PR, revert them here, and do the `EnablePreviewFeatures` workaround.

lewing replied:
It is unlikely that we can change the target here in a single pass, the sdk typically takes months to adjust

jkoritzinsky replied:
I think we're actually pretty close to updating the TFM. Looks like we need to update the bootstrapping scripts for the installer tests and a little bit of debugging of the WBT legs.

Next week I'll branch off the upgrading to another PR and do the quick workaround for the EnablePreviewFeatures attribute to unblock this PR.

jkoritzinsky committed:
Remove explicit hard-coding of net6.0 package testing

jkoritzinsky committed:
Convert more references to net8.0 to net9.0

ericstj commented on Directory.Build.props:
```diff
@@ -80,20 +80,20 @@
 
     <!-- The minimum supported .NET version. -->
-    <NetCoreAppMinimum>net6.0</NetCoreAppMinimum>
+    <NetCoreAppMinimum>net8.0</NetCoreAppMinimum>
```
We still support net6.0 for another year - are we really ready to move this?  Also I expect we will have projects that expect Previous and Minimum to be different.  @ViktorHofer

jkoritzinsky replied:
Doesn't .NET 6.0 go out of support on the day that .NET 9 comes out?

ViktorHofer replied:
We did the same when we switched from .NET 6 to .NET 7: We removed `netcoreapp3.1` and `net5.0` support as when .NET 7 released, those runtimes were already officially out-of-support.

That model worked well with .NET 7 and I would prefer to drop the TFMs now than in the middle of .NET 9 development.

> Also I expect we will have projects that expect Previous and Minimum to be different.

We should fix such wrong assumptions. I don't recall any managed project that requires those two properties to differ. Those were designed to be equal every odd major release version. And, source build already sets all three properties to the same value.

ViktorHofer commented on eng/Analyzers.targets:
```diff
@@ -1,4 +1,8 @@
+  <PropertyGroup Condition="'$(UsingMicrosoftNoTargetsSdk)' == 'true' or '$(UsingMicrosoftDotNetSharedFrameworkSdk)' == 'true'">
+    <!-- Explicitly disable running analyzers to avoid trying to discover the correct ILLink tool pack for a project that has no sources. -->
+    <RunAnalyzers>false</RunAnalyzers>
+  </PropertyGroup>
```
nit: Personally I would prefer this to be a condition directly on the property instead of defining a new group but that's just my own taste of style. We did the same for the DotNetBuildFromSource condition below...

ViktorHofer commented:
To fix the installer failures, you need to update https://github.com/dotnet/runtime/blob/9ddb7316837cafecceca637dabb1a9b1f2b08409/src/installer/tests/Assets/TestUtils/TestProjects.targets#L7-L31

lewing committed:
Merge branch 'main' into BumpMainTo90Alpha1

BrennanConroy committed:
Fix queue count in rate limiters (#90810)

jkoritzinsky committed:
Revert TFM changes

radical commented:
I'm looking at the wasm failures now.

radical committed:
[wasm] Unpin sdk version, and change channel to 9.0 for WBT

radical committed:
[wasm] Unpin sdk version, and change channel to 9.0 for WBT

jkoritzinsky committed:
Add workaround for RequiresPreviewFeatures and regenerate SLE compat suppressions.

jkoritzinsky commented:
I've reverted the TFM upgrade to avoid bundling together too many different concepts in this PR, as requested.

I'll open a new draft PR in a minute with the TFM changes. @radical I'll revert your commit here and cherry-pick it to the other PR.

jkoritzinsky committed:
Revert "[wasm] Unpin sdk version, and change channel to 9.0 for WBT"

This reverts commit be2c615226a6b3a53bfab056c1b7cb7c150d2f48.

jkoritzinsky commented:
PR with the TFM upgrade is at https://github.com/dotnet/runtime/pull/90880

carlossanlop commented:
> PR with the TFM upgrade is at #90880

Thank you so much, @jkoritzinsky, that is a huge contribution. I think it's best to send it separately.

HongGit committed:
Adding PACKAGE.md to System.Runtime.Caching package (#90701)

* Adding PACKAGE.md to System.Runtime.Caching package

* add Remarks

* move PACKAGE.md to src folder

* update per PR feedback

* add important block

jkoritzinsky commented:
> > PR with the TFM upgrade is at #90880
> 
> Thank you so much, @jkoritzinsky, that is a huge contribution. I think it's best to send it separately.

I agree. This PR only has the changes required for the rebranding, not the TFM upgrade.

ViktorHofer commented on src/libraries/System.Linq.Expressions/src/CompatibilitySuppressions.xml:
```diff
@@ -139,6 +139,48 @@
     <Left>ref/net8.0/System.Linq.Expressions.dll</Left>
     <Right>lib/net8.0/System.Linq.Expressions.dll</Right>
   </Suppression>
+  <Suppression>
```
These changes should be reverted

jkoritzinsky replied:
The build fails without these two changes.

jkoritzinsky committed:
Suppress EventLog package validation failures to get allconfigurations passing.

radical commented:
New sdk error on windows:
```
EXEC : error : dotnet-install: Failed to check the disk space. Installation will continue, but it may fail if you do not have [D:\a\_work\1\s\src\mono\wasm\Wasm.Build.Tests\Wasm.Build.Tests.csproj]
##[error]EXEC(0,0): error : (NETCORE_ENGINEERING_TELEMETRY=Build) dotnet-install: Failed to check the disk space. Installation will continue, but it may fail if you do not have
   enough disk space.
  dotnet-install: Extracting the archive.
```

The log doesn't show the exit code. I'll check locally.

jkoritzinsky committed:
Update Version.Details.xml for xlifftasks to try to fix prebuilt error.

radical commented:
> New sdk error on windows:
> 
> ```
> EXEC : error : dotnet-install: Failed to check the disk space. Installation will continue, but it may fail if you do not have [D:\a\_work\1\s\src\mono\wasm\Wasm.Build.Tests\Wasm.Build.Tests.csproj]
> ##[error]EXEC(0,0): error : (NETCORE_ENGINEERING_TELEMETRY=Build) dotnet-install: Failed to check the disk space. Installation will continue, but it may fail if you do not have
>    enough disk space.
>   dotnet-install: Extracting the archive.
> ```
> 
> The log doesn't show the exit code. I'll check locally.

I'm not sure what is wrong here. The `dotnet-install.ps1` script uses `Write-Warning` to emit the message, and this script is being run with `Exec` msbuild task with `IgnoreStandardErrorWarningFormat="true"`. I don't know how it is becoming an error. The task doesn't even seem to fail the build, and instead the whole build completes, and then msbuild shows the error at the end.

ViktorHofer reviewed (commented):
The AssemblyVersion property in Versions.props must be hardcoded to 8.0.0.0 for now. We had the same issue last release and noticed that updating the TFM must happen in conjunction with reving the major assembly version.

Please also undo the changes in both CompatibilitySuppressions.xml, the TestData.resources one and the PACKAGE.md change which looks unrelated.

radical committed:
workloads-testing.targets: Ignore standard error/warning formats

Prompted by the following getting caught as a warning, and converted to
an error:

```
EXEC : error : dotnet-install: Failed to check the disk space. Installation will continue, but it may fail if you do not have [D:\a\_work\1\s\src\mono\wasm\Wasm.Build.Tests\Wasm.Build.Tests.csproj]
   enough disk space.
  dotnet-install: Extracting the archive.
```

jkoritzinsky committed:
Force AssemblyVersion

jkoritzinsky committed:
Revert changes.

jkoritzinsky committed:
Remove the SLE suppressions again.

jkoritzinsky committed:
Fix 9.0->8.0 replace

jkoritzinsky committed:
Revert "Fix queue count in rate limiters (#90810)"

This reverts commit 1b9e4e3c7e80ac20877e5172940f32acdf346adf.

jkoritzinsky committed:
Fix formatting

radical commented:
/azp run runtime-wasm

jkoritzinsky commented:
@radical if the WASM MT failures are known, can you merge this in?
